src/qt/walletview.cpp

```diff
@@ -222,6 +225,69 @@ void WalletView::gotoVerifyMessageTab(QString addr)
         signVerifyMessageDialog->setAddress_VM(addr);
 }
 
+void WalletView::gotoLoadPSBT()
+{
+    QString filename = GUIUtil::getOpenFileName(this,
+        tr("Load Transaction Data"), QString(),
+        tr("Partially Signed Transaction (*.psbt)"), nullptr);
+    if (filename.isEmpty()) return;
+    std::ifstream in(filename.toLocal8Bit().data(), std::ios::binary);
+    // https://stackoverflow.com/questions/116038/what-is-the-best-way-to-read-an-entire-file-into-a-stdstring-in-c
+    std::string data(std::istreambuf_iterator<char>{in}, {});
+
+    std::string error;
+    PartiallySignedTransaction psbtx;
+    if (!DecodeRawPSBT(psbtx, data, error)) {
+        Q_EMIT message(tr("Error"), tr("Unable to decode PSBT file") + "\n" + QString::fromStdString(error), CClientUIInterface::MSG_ERROR);
+        return;
+    }
+
+    PSBTAnalysis analysis = AnalyzePSBT(psbtx);
+    bool have_all_sigs = (analysis.next == PSBTRole::FINALIZER) || (analysis.next == PSBTRole::EXTRACTOR);
+
+    CMutableTransaction mtx;
+    bool complete = FinalizeAndExtractPSBT(psbtx, mtx);
+
+    QMessageBox msgBox;
+    msgBox.setText("Partially Signed Bitcoin Transaction");
+
+    if (complete) {
+        msgBox.setInformativeText("Would you like to send this transaction?");
+    } else if (!have_all_sigs) {
+        msgBox.setInformativeText("Transaction needs more signatures. Copy to clipboard?");
+    } else {
+        msgBox.setInformativeText("Transaction is incomplete. Copy to clipboard?");
```

Comment:
what situations can this hit?

Reply:
From my own study of this question, I believe the the answer is "no situations, unless there are bugs in AnalyzePSBT", but I think we've run into some bugs in AnalyzePSBT while figuring this out.

Reply:
I don't feel confident enough about `AnalyzePSBT` (yet) to put an assert here.

Reply:
Worst thing I'd do is stick a log line in debug or something. Fine without.

Reply:
I believe this can be hit if there are `final_scriptSig`/`final_scriptWitness` but with invalid signatures. So `have_all_sigs == true` because there are sigs there (it does not validate), but `complete == false` because it does validate and finds the sigs to be invalid.

Some things that could happen where we might want better error messages:

* Logically invalid psbt (see the conditions where `next` is `Creator`)
* Missing things, in general needs updating rather than signing.

Reply:
Actually I just tested this case and it seems that because it is already finalized, the `final_scriptSig`/`final_scriptWitness` is assumed to be valid so `complete == true`. So what we actually get is a failed attempt to broadcast. But there's no error message, just a blank error dialog.

I think in general for this section, we should do the analysis first and do a `switch` on the result to give more specific errors and messages about the analysis. Then if finalize/extract is the next step, do that. It may also be prudent to have `FinalizeAndExtractPSBT` to always verify.